packages/indexer-common/src/operator.ts

```diff
@@ -70,152 +62,38 @@ const disputeFromGraphQL = (
   return obj as POIDisputeAttributes
 }
 
-interface indexNode {
-  id: string
-  deployments: string[]
-}
-
-export class Indexer {
-  statusResolver: IndexingStatusResolver
-  rpc: RpcClient
-  indexerManagement: IndexerManagementClient
+// This component is responsible for managing indexing rules, actions, allocations, and
+// POI disputes.
+export class Operator {
   logger: Logger
-  indexNodeIDs: string[]
-  defaultAllocationAmount: BigNumber
-  indexerAddress: string
-  allocationManagementMode: AllocationManagementMode
+  indexerManagement: IndexerManagementClient
+  specification: spec.NetworkSpecification
 
   constructor(
     logger: Logger,
-    adminEndpoint: string,
-    statusResolver: IndexingStatusResolver,
     indexerManagement: IndexerManagementClient,
```

Comment:
Does it make sense to generalize some of the queries made to `indexerManagementClient` field so that not every operator own a client? or if there exists a query that's network specific and forces operator to require indexer management access?

Reply:
I really want to decouple the `IndexerManagementClient` from the `Operator` class. 
I ended up with a dependency cycle between them and the `Network` class, which I couldn't resolve.

Ideally, the IndexerManagementClient should be a separate service, but I believe this will be a whole new endeavour.